### PR TITLE
INSTALL-944: Apply latest enhancements to the CI pipeline

### PR DESCRIPTION
**Pull request by lulivi**
### Checks

-   [x] I have updated the documentation accordingly.
-   [x] I have read the [CONTRIBUTING](https://github.com/rticommunity/rticonnextdds-examples/blob/develop/CONTRIBUTING.md) document.

### CODE REVIEW

File: resources/docker/Dockerfile.linux [hunk 1]
```diff
@@ -11,20 +11,18 @@
...
 
-ENV CONNEXTDDS_ARCH="x64Linux4gcc7.3.0"
+ARG USER_NAME=jenkins
+ARG USER_UID
+
 ENV DEBIAN_FRONTEND="noninteractive"
+ENV CONNEXTDDS_ARCH="x64Linux4gcc7.3.0"
 
 RUN apt-get update && apt-get install -y \
-    gcc \
```
Comment (ManuelJNunez) [inline resources/docker/Dockerfile.linux:18]: Isn't the multiline better? https://docs.docker.com/develop/develop-images/instructions/#apt-get

**Comment by ManuelJNunez** [comment]
@lulivi would you like to use this PR to remove the python scripts?

File: resources/docker/Dockerfile.linux (hunk 1 above)
Comment (lulivi) [reply resources/docker/Dockerfile.linux:18]: Sure, why not?

**Comment by lulivi** [comment]
The validation errors will be gone once we update the cmake-utils submodule with the unmerged latest changes

File: Jenkinsfile [hunk 2]
```diff
@@ -10,33 +10,16 @@
...
- * @param linkMode The link mode name.
- */
-void runBuildStage(String buildMode, String linkMode) {
-    String cmd = "python3 ${env.WORKSPACE}/resources/ci_cd/linux_build.py"
-    cmd += " --build-mode ${buildMode}"
-    cmd += " --link-mode ${linkMode}"
-    cmd += " --build-dir ${getBuildDirectory(buildMode, linkMode)}"
-    command.run(cmd)
-}
+@Library("rticommunity-jenkins-pipelines@feature/INSTALL-944") _
```
Comment (lulivi) [inline Jenkinsfile:13]: Remove this line after the shared library PR is merbed

**Review by ManuelJNunez — APPROVED** [review]

File: Jenkinsfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (lulivi) [reply Jenkinsfile]: Its done!

**Pull request merged**